### PR TITLE
retrieval: Handle serverset node not existing.

### PR DESCRIPTION
**Pull request by brian-brazil**
This stops configuration loading hanging if
the Znode doesn't exist, and retries until the
node does exist.

### CODE REVIEW

File: retrieval/discovery/serverset.go [hunk 1]
```diff
@@ -279,16 +280,20 @@ func (tc *zookeeperTreeCache) loop(failureMode bool) {
...
 			log.Infof("Attempting to resync state with Zookeeper")
 			err := tc.recursiveNodeUpdate(tc.prefix, tc.head)
-			if err == nil {
-				failureMode = false
-			} else {
+			if err != nil {
 				log.Errorf("Error during Zookeeper resync: %s", err)
 				failure()
+			} else {
+				log.Infof("Zookeeper resync successful")
```
Comment (fabxc) [inline retrieval/discovery/serverset.go]: This is a candidate for `Debugf`

Comment (brian-brazil) [inline retrieval/discovery/serverset.go]: I added it as it wasn't at all obvious from the logs if a resync had worked, or stalled somehow.

**Comment by fabxc** [comment]
:+1: otherwise

File: retrieval/discovery/serverset.go (hunk 1 above)
Comment (fabxc) [inline retrieval/discovery/serverset.go]: Not feeling too strongly as long as it is not a spammy log.

Comment (brian-brazil) [inline retrieval/discovery/serverset.go]: Generally it should only happen after zookeeper recovers from a failure, the above error will be more common.

Comment (fabxc) [inline retrieval/discovery/serverset.go]: Fine with me then.

**Comment by brian-brazil** [comment]
Thanks.

**Pull request merged**